### PR TITLE
ADAPTSM-150 | @rebeccahongsf | add membership form flow for friend/non-alumni

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- add membership form flow for friend/non-alumni

# Review By (Date)
- asap

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch and spin up your local with mockserver enabled (`MEGAPROFILE_MOCK = 'true'`)
2. Navigate to `/membership/register`
3. Select `Myself`. The payment options should not display
5. Continue to the form
6. Confirm that the form displays the following two options:
image

7. Now we'll test for `zguan` through the preview (or disable your mockserver)
8. Login as `zguan`
9. Navigate to `/membership/register`
10. Select `Myself`. The payment options should display
11. Select `Installment`
12. Continue to form
13. Confirm that the installments form displays:
image

14. Navigate to back to `/membership/register`
15. Select `Myself`. The payment options should display
16. Select `Full payment`
17.  Continue to form
18. Confirm that the full payment form displays:
image

19. Review code 🐛 

# Associated Issues and/or People
- ADAPTSM-150

### CODE REVIEW

File: src/utilities/mocks/affiliations.js [hunk 1]
```diff
@@ -8,10 +8,11 @@ export const affiliationsMockData = {
     },
   ],
   affiliations: [
```
Comment (rebeccahongsf) [inline src/utilities/mocks/affiliations.js:10]: @rebeccahongsf to revert this change before merge!

**Review by jdwjdwjdw — APPROVED** [review]
Everything's working as specified, thanks @rebeccahongsf. Great job catching this scenario 🎉
